storage/series/series.go

```diff
@@ -401,26 +401,33 @@ func (s *dbSeries) bufferDrained(newBlock block.DatabaseBlock) {
 	// lock already. Executing the drain method occurs during a write if the
 	// buffer needs to drain or if tick is called and series explicitly asks
 	// the buffer to drain ready buckets.
-	s.mergeBlockWithLock(s.blocks, newBlock)
+	iOpts := s.opts.InstrumentOptions()
+	scope := iOpts.MetricsScope().SubScope("series-buffer-drain")
+	err := s.mergeBlockWithLock(newBlock)
+	if err != nil {
+		iOpts.Logger().WithFields(
+			xlog.NewField("id", s.id.String()),
+			xlog.NewField("blockStart", newBlock.StartTime()),
+			xlog.NewField("err", err.Error()),
+		).Errorf("error trying to drain series buffer")
+		scope.Counter("buffer-drain-error").Inc(1)
```

Comment:
Might be best to cache the `logger xlog.Logger` on the series struct so you can just `s.log.Errorf(...)` and also create a `type seriesMetrics { bufferDrainError tally.Counter }` to make it consistent with other types of metric structs across the code base.

Reply:
Spoke offline, decided not to do this to avoid adding extra fields to series struct (million in memory at a time)